rs/tests/networking/canister_http_stress_test.rs

```diff
@@ -0,0 +1,366 @@
+                    concurrent_requests: *concurrent_requests,
+                    qps,
+                    average_latency_s: duration_s,
+                });
+            }
+            leave_proxy_canister_running(&proxy_canister, url.clone(), logger.clone()).await;
+        });
+    }
+    let base_dir = env.base_path();
+    let json_file = base_dir.join(BENCHMARK_REPORT_FILE);
+    std::fs::create_dir_all(json_file.parent().unwrap()).unwrap();
+
+    let json_str = serde_json::to_string_pretty(&all_results).unwrap();
+    std::fs::write(&json_file, json_str).unwrap();
+
+    info!(
+        logger,
+        "All benchmark results have been written to {:?}", json_file
+    );
+}
+
+async fn leave_proxy_canister_running(proxy_canister: &Canister<'_>, url: String, logger: Logger) {
+    ic_system_test_driver::retry_with_msg_async!(
+        format!(
+            "calling start_continuous_requests of proxy canister {} with URL {}",
+            proxy_canister.canister_id(),
+            url
+        ),
+        &logger,
+        READY_WAIT_TIMEOUT,
+        RETRY_BACKOFF,
+        || async {
+            let context = "There is context to be appended in body";
+            let res = proxy_canister
+                .update_(
+                    "start_continuous_requests",
+                    candid_one::<
+                        Result<RemoteHttpResponse, (RejectionCode, String)>,
+                        RemoteHttpRequest,
+                    >,
+                    RemoteHttpRequest {
+                        request: UnvalidatedCanisterHttpRequestArgs {
+                            url: url.to_string(),
+                            headers: vec![],
+                            body: None,
+                            transform: Some(TransformContext {
+                                function: TransformFunc(candid::Func {
+                                    principal: proxy_canister.canister_id().get().0,
+                                    method: "transform_with_context".to_string(),
+                                }),
+                                context: context.as_bytes().to_vec(),
+                            }),
+                            method: HttpMethod::GET,
+                            max_response_bytes: None,
+                        },
+                        cycles: 500_000_000_000,
+                    },
+                )
+                .await
+                .expect("Update call to proxy canister failed");
+            if !matches!(res, Ok(ref x) if x.status == 200) {
+                bail!("Http request failed response: {:?}", res);
+            }
+            info!(&logger, "Update call succeeded! {:?}", res);
+            Ok(())
+        }
+    )
+    .await
+    .expect("Timeout on doing a canister http call to the webserver");
+}
+
+async fn do_request(
+    proxy_canister: &Canister<'_>,
+    url: String,
+    logger: &Logger,
+    concurrent_requests: u64,
+) -> Result<u64, anyhow::Error> {
```

Comment:
maybe it would be easier to use if we return `std::time::Duration` here?
```suggestion
) -> Result<Duration, anyhow::Error> {
```

Reply:
Done, much better. thanks